src/common/ad_tx.c

```diff
@@ -881,11 +1045,24 @@ static int
 umo_tx_publish(struct umem_instance *umm, void *actv, int actv_cnt)
 {
 	struct ad_tx		*tx = tx_get();
+	struct ad_reserv_act	*ractv = actv;
+	int			 i, rc;
 
 	D_ASSERTF(ad_tx_stage(tx) == UMEM_STAGE_WORK,
 		  "TX "DF_U64", bad stage %d\n", ad_tx_id(tx), ad_tx_stage(tx));
 
-	return ad_tx_publish(tx, (struct ad_reserv_act *)actv, actv_cnt);
+	rc = ad_tx_publish(tx, ractv, actv_cnt);
+	if (rc == 0) {
+		for (i = 0; i < actv_cnt; i++) {
+			rc = tx_range_add(tx, ractv[i].ra_off, ractv[i].ra_size, true);
```

Comment:
I'm not sure if we should always do this, in the case of DTX (or. btree, evtree), we can allocate a piece of memory, which is a few kilo/hundreds bytes, and populate tens of bytes only. In this case, adding tens of bytes to redo log is sufficient, instead of hundreds.

This is the common issue of ad_mem and bmem, we might have to change the current code to make sure we only add required number of bytes to redo log.

Reply:
We only use 'reserve' interface for values, for VOS index, we always allocate by tx_alloc(), so if the 'publish' here is only for values (allocated through reserve interface),  I think we'll be good?